### PR TITLE
feat: auto-tracking session events

### PR DESCRIPTION
**Pull request by falconandy**
### Summary

1. Added new option (`options.trackingOptions.sessionEvents`) to auto-track session start/end events.
2. Added tests for session logic.

Tested only on Android emulator.

Existing differences between `Amplitude-Kotlin` and `React Native` implementations:
1. `React Native` has ability to set custom session id (client `options.sessionId`, `setSessionId`), `Amplitude-Kotlin` doesn't have.
2. `Amplitude-Kotlin` persists last event Id, `React Native` doesn't - always starts event Id from `0` on app restart.
3. Session timeout is 5 minutes in `Amplitude-Kotlin` and 30 minutes in `React Native`.

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No, but session logic could be affected.

### CODE REVIEW

File: packages/analytics-types/src/config.ts [hunk 1]
```diff
@@ -76,6 +76,7 @@ export interface TrackingOptions {
 export interface ReactNativeTrackingOptions extends TrackingOptions {
   adid?: boolean;
   carrier?: boolean;
+  sessionEvents?: boolean;
```
Comment (kevinpagtakhan) [inline packages/analytics-types/src/config.ts:79]: I think this is called `trackingSessionEvents` in Kotlin, perhaps we can align?

Comment (falconandy) [reply packages/analytics-types/src/config.ts:79]: @kevinpagtakhan 
Do you mean `options.trackingSessionEvents` (as in Kotlin) vs `options.trackingOptions.trackingSessionEvents`?

Comment (kevinpagtakhan) [reply packages/analytics-types/src/config.ts:79]: this one: `options.trackingSessionEvents`

Comment (falconandy) [reply packages/analytics-types/src/config.ts]: Fixed to `options.trackingSessionEvents`. Currently in `ReactNativeConfig`, in the future can be moved to base `BrowserConfig` if session events auto-tracking will be implemented in browsers too

**Review by qingzhuozhen — APPROVED** [review]
Test on iOS and can see the session events. Will create tickets to handle the event id and default session.

**Pull request merged**